src/com/java/multithreading/BlockingQueue.java

```diff
@@ -0,0 +1,46 @@
+package com.java.multithreading;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class BlockingQueue {
+
+  private Queue<Integer> queue;
+  private int capacity;
+
+  public BlockingQueue(int capacity) {
+    this.queue = new LinkedList<Integer>();
+    this.capacity = capacity;
+  }
+
+  public synchronized  void enqueue(Integer item){
+    while(queue.size() == capacity){
+      try {
+        wait();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    }
+    queue.add(item);
+    System.out.println("Produced "+item);
+    notify();
```

Comment:
Why did you used notify() and not notifyAll()?..I was asked this question in an interview?
Which one is better here?..notify or notifyAll?